src/hypofuzz/hypofuzz.py

```diff
@@ -511,6 +555,20 @@ def fuzz_several(*targets_: FuzzProcess, random_seed: Optional[int] = None) -> N
     for target in targets:
         target.startup()
 
+    dispatch = {}
+    for target in targets:
+        # each node only gets one fuzz process
+        assert target.database_key not in dispatch
+        dispatch[target.database_key] = target
```

Comment:
I Regret To Inform You, this is usually but not strictly always true (eg iirc if you set the `__name__` of a function Pytest probably still mostly works??).

I wouldn't bother if it was difficult to support, but accumulating a dict of lists and iterating over the callbacks seems equally easy to implement (`if event := ...: for callback in dispatch.get(event.database_key, []): ...`), etc.

Reply:
hmm, if you have *database key* collisions, I think you're going to have a bad time regardless! Shared corpus isn't so bad, but shared observations, failures, and reports is.

That said, as you say this is easy to support, and we can think about other ramifications of database key collisions later.

Reply:
Oops, I was thinking of `nodeid` collisions (hence mentioning pytest), but it's _also_ true that tests can have database collisions; we try to avoid it but it _can and does_ occasionally happen in practice.

You're right that this is not a great time, but I think we do ~okay just by making sure that we handle it as best we reasonably can in each place that comes up outside of the literal database collisions.